sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+"""Implementations of pathways for use by actuators.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import S, Symbol
+from sympy.core.expr import Expr
+from sympy.physics.mechanics import Point
+from sympy.physics.vector import Vector, dynamicsymbols
+
+
+__all__ = ['LinearPathway']
+
+
+class PathwayBase(ABC):
+    """Abstract base class for all pathway classes to inherit from.
+
+    Notes
+    =====
+
+    Instances of this class cannot be directly instantiated by users. However,
+    it can be used to created custom pathway types through subclassing.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``PathwayBase``."""
+        self.origin = origin
+        self.insertion = insertion
+
+    @property
+    def origin(self) -> Point:
+        """The ``Point`` from which the pathway originates."""
+        return self._origin
+
+    @origin.setter
+    def origin(self, origin: Point) -> None:
+        if not isinstance(origin, Point):
+            msg = (
+                f'Value {repr(origin)} passed to `origin` was of type '
+                f'{type(origin)}, must be {type(Point)}.'
+            )
+            raise TypeError(msg)
+        self._origin = origin
+
+    @property
+    def insertion(self) -> Point:
+        """The ``Point`` into which the pathway inserts."""
+        return self._insertion
+
+    @insertion.setter
+    def insertion(self, insertion: Point) -> None:
+        if not isinstance(insertion, Point):
+            msg = (
+                f'Value {repr(insertion)} passed to `insertion` was of type '
+                f'{type(insertion)}, must be {type(Point)}.'
+            )
+            raise TypeError(msg)
+        self._insertion = insertion
+
+    @abstractmethod
+    def _true_length(self) -> Expr:
```

Comment:
Maybe `_get_length` or `_compute_length`? I would prefer the first one.

Reply:
See other thread on the `_true_length` method. If you're in agreement with my points there then I'd argue this method is well named.

Reply:
Addressed in https://github.com/sympy/sympy/pull/24914/commits/14b28b40e8ab3f6a82cfed08ef3b33aae095db6f.